scwx-qt/source/scwx/qt/map/marker_layer.cpp

```diff
@@ -0,0 +1,108 @@
+#include <scwx/qt/map/marker_layer.hpp>
+#include <scwx/qt/manager/marker_manager.hpp>
+#include <scwx/util/logger.hpp>
+#include <scwx/qt/types/marker_types.hpp>
+#include <scwx/qt/types/texture_types.hpp>
+#include <scwx/qt/gl/draw/geo_icons.hpp>
+
+namespace scwx
+{
+namespace qt
+{
+namespace map
+{
+
+static const std::string logPrefix_ = "scwx::qt::map::marker_layer";
+static const auto        logger_    = scwx::util::Logger::Create(logPrefix_);
+
+class MarkerLayer::Impl
+{
+public:
+   explicit Impl(MarkerLayer* self, std::shared_ptr<MapContext> context) :
+       self_ {self}, geoIcons_ {std::make_shared<gl::draw::GeoIcons>(context)}
+   {
+      ConnectSignals();
+   }
+   ~Impl() {}
+
+   void ReloadMarkers();
+   void ConnectSignals();
+
+   MarkerLayer* self_;
+   const std::string& markerIconName_ {
+      types::GetTextureName(types::ImageTexture::Cursor17)};
```

Comment:
Eventually recommend a unique icon.  I've used Font Awesome Free and Flaticon.

Reply:
That is defiantly just a temporary placeholder. I was planning on having a few selectable icons. Thanks for recommending a place to start looking.